### ISSUE TITLE
Pretty.h issue

### ISSUE THREAD

**Issue by detective7th** [facebook/folly#1247]
tool:
gcc 9.2.0 with c++14 c++17 c++20

code:
#include <folly/lang/Pretty.h>
auto b = folly::pretty_name<uint16_t>();

compilation message:
/usr/local/include/folly/lang/Pretty.h: In instantiation of ‘struct folly::detail::pretty_carray<18446744073709551615>’:
/usr/local/include/folly/lang/Pretty.h:137:44:   required from ‘struct folly::detail::pretty_name_zarray<short unsigned int, folly::detail::pretty_tag_gcc>’
/usr/local/include/folly/lang/Pretty.h:156:76:   required from ‘constexpr const char* folly::pretty_name() [with T = short unsigned int]’
/home/azi/workspace/flash/flash/relay/src/main.cpp:40:38:   required from here
/usr/local/include/folly/lang/Pretty.h:30:8: error: size ‘18446744073709551615’ of array exceeds maximum object size ‘9223372036854775807’
   30 |   char data[S];
      |        ^~~~
/usr/local/include/folly/lang/Pretty.h: In instantiation of ‘constexpr const folly::detail::pretty_carray<18446744073709551615> folly::detail::pretty_name_zarray<short unsigned int, folly::detail::pretty_tag_gcc>::zarray’:
/usr/local/include/folly/lang/Pretty.h:156:76:   required from ‘constexpr const char* folly::pretty_name() [with T = short unsigned int]’
/home/azi/workspace/flash/flash/relay/src/main.cpp:40:38:   required from here
/usr/local/include/folly/lang/Pretty.h:137:60:   in ‘constexpr’ expansion of ‘folly::detail::pretty_name_zarray<short unsigned int, folly::detail::pretty_tag_gcc>::zarray_()’
/usr/local/include/folly/lang/Pretty.h:137:44: error: array subscript value ‘1’ is outside the bounds of array type ‘char [1]’
  137 |   static constexpr pretty_carray<size + 1> zarray = zarray_();

**Comment by georgios-v** [comment]
@detective7th hey, I have the same error. How did you solve it?

**Comment by yfeldblum** [comment]
I am not sure how to reproduce this.

What is the content of the array in:

```lang=c++
auto arr = folly::detail::pretty_raw<uint16_t>(
    folly::detail::pretty_tag_gcc{});
```

**Comment by georgios-v** [comment]
I have _worked around_ this issue so I now know what caused it:

Let me start by saying that proxygen compiled and all tests and samples run normally.

I was integrating proxygen with my own codebase, an existing codebase, when I got this error. My codebase was already using a logger, g3log, which I did not remove before the integration. The error was caused by every unit that included `<g3log/g3log.h>` before any proxygen include. By moving the proxygen includes before any other the problem disappeared. Thus, I can infer that it was caused by a macro or a define in g3log.
